app/scripts/router.js

```diff
@@ -80,11 +80,32 @@ function (
 
   function showView(View, options) {
     return function () {
-      this.createAndShowView(View, options);
+      // If the current view is an instance of View, that means we're
+      // navigating from a subview of the current view
+      if (this.currentView instanceof View) {
+        this.currentView.trigger(this.NAVIGATE_FROM_SUBVIEW, options);
+      } else {
+        this.createAndShowView(View, options);
+      }
+    };
+  }
+
+  // Show a sub-view, creating and initializing the SuperView if needed.
+  function showSubView(SuperView, options) {
+    return function () {
+      // If currentView is of the SuperView type, simply show the subView
+      if (this.currentView instanceof SuperView) {
+        this.currentView.showSubView(options.subView, options);
+      } else {
+        // Create the SuperView; its initialization method should handle the subView option.
+        this.createAndShowView(SuperView, options);
```

Comment:
We already use the [Subview](https://github.com/mozilla/fxa-content-server/blob/719b2ae68ecb004f3946948f516bd1de0293ef5f/app/scripts/views/base.js#L370), so maybe leave that the same.